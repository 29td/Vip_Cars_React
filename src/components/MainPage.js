import { useEffect, useRef } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faTriangleExclamation } from '@fortawesome/free-solid-svg-icons';
import { Swiper, SwiperSlide } from 'swiper/react';
import {
  Navigation,
  Pagination, A11y,
} from 'swiper';
import { allCars, getAllCars } from '../redux/cars/carSlice';
import Navbar from './Navbar';
import 'swiper/css';
import 'swiper/css/navigation';
import 'swiper/css/pagination';

const MainPage = () => {
  const cars = useSelector(allCars);

  const dispatch = useDispatch();
  const swiperRef = useRef();
  const navigate = useNavigate();

  useEffect(() => { dispatch(getAllCars()); }, [dispatch]);

  return (
    <div className="App">
      <Navbar />
      <main className="main main-page">
        { cars.length < 1 ? (
          <section className="no-res">
            <FontAwesomeIcon icon={faTriangleExclamation} className="excl" />
            <h2>No Car has been made yet!</h2>
            <p>Add a car using the button bellow</p>
            <button className="link" type="button" onClick={() => navigate('/add_car')}>Add Car</button>
          </section>
        ) : (
          <>
            <h1 className="title">LATEST VIP CARS</h1>
            <p className="sub-title">Please select a Car to Reserve</p>

            <section className="cars-container">
              <Swiper
                modules={[Navigation, Pagination, A11y]}
                pagination={{ clickable: true }}
                scrollbar={{ draggable: true }}
                onBeforeInit={(swiper) => {
                  swiperRef.current = swiper;
                }}
                breakpoints={{
                  // when window width is >= 640px
                  0: {
                    slidesPerView: 1,
                  },
                  // when window width is >= 768px
                  768: {
                    spaceBetween: 15,
                    slidesPerView: 2,
                  },
                  // when window width is >= 768px
                  900: {
                    spaceBetween: 20,
                    slidesPerView: 3,
                  },
                }}
                className="mySwiper"
              >
                { cars.map((car) => (
                  <SwiperSlide key={car.id}>
                    <a href={`/cars/${car.id}`} className="car">
                      <div className="car-img-wrap">
                        <img className="car-img" src={car.image} alt={car.name} />
                      </div>
                      <div className="car-details">
                        <h3 className="car-name">
                          {car.name.slice(0, 20)}
                          {car.name.length > 20 ? '...' : ''}
                        </h3>
                        <p className="car-model">{car.model}</p>
                        <hr />
                        <p className="car-desc">
                          {car.description.slice(0, 50)}
                          {car.description.length > 50 ? '...' : ''}
                        </p>
                      </div>
                    </a>
                  </SwiperSlide>
                ))}
              </Swiper>
              <button type="button" aria-label="previous" className="btn prev" onClick={() => swiperRef.current?.slidePrev()} />
              <button type="button" aria-label="previous" className="btn next" onClick={() => swiperRef.current?.slideNext()} />
            </section>
          </>
        ) }
      </main>
    </div>
  );
};

export default MainPage;
