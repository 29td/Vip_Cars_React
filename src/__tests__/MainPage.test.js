import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/';
import { BrowserRouter as Router } from 'react-router-dom';
import { AuthProvider } from 'react-auth-kit';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import MainPage from '../components/NavFooter';

describe('Login component', () => {
  test('Should render the proper Login component', () => {
    const login = render(
      <React.StrictMode>
        <Provider store={store}>
          <AuthProvider
            authType="cokie"
            authName="_auth"
            cookieDomain={window.location.hostname}
            cookieSecure={false}
          >
            <Router>
              <MainPage />
            </Router>
          </AuthProvider>
        </Provider>
        ,
      </React.StrictMode>,
    );
    expect(login).toMatchSnapshot();
  });
});
