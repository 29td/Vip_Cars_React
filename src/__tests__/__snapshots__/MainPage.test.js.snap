// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component Should render the proper Login component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="NavFooter"
      >
        <footer
          class="text-center text-white"
        >
          <div
            class="container pt-4"
          >
            <section
              class="mb-4"
            >
              <a
                class="btn btn-link btn-floating btn-lg text-dark"
                data-mdb-ripple-color="dark"
                href="#!"
                role="button"
              >
                <i
                  class="fa fa-twitter"
                />
              </a>
              <a
                class="btn btn-link btn-floating btn-lg text-dark"
                data-mdb-ripple-color="dark"
                href="#!"
                role="button"
              >
                <i
                  class="fa fa-facebook-f"
                />
              </a>
              <a
                class="btn btn-link btn-floating btn-lg text-dark"
                data-mdb-ripple-color="dark"
                href="#!"
                role="button"
              >
                <i
                  class="fa fa-google-plus"
                />
              </a>
              <a
                class="btn btn-link btn-floating btn-lg text-dark"
                data-mdb-ripple-color="dark"
                href="#!"
                role="button"
              >
                <i
                  class="fa fa-vimeo"
                />
              </a>
              <a
                class="btn btn-link btn-floating btn-lg text-dark"
                data-mdb-ripple-color="dark"
                href="#!"
                role="button"
              >
                <i
                  class="fa fa-pinterest"
                />
              </a>
              <a
                class="btn btn-link btn-floating btn-lg text-dark"
                data-mdb-ripple-color="dark"
                href="#!"
                role="button"
              >
                <i
                  class="fa fa-github"
                />
              </a>
            </section>
          </div>
        </footer>
      </div>
      ,
    </div>
  </body>,
  "container": <div>
    <div
      class="NavFooter"
    >
      <footer
        class="text-center text-white"
      >
        <div
          class="container pt-4"
        >
          <section
            class="mb-4"
          >
            <a
              class="btn btn-link btn-floating btn-lg text-dark"
              data-mdb-ripple-color="dark"
              href="#!"
              role="button"
            >
              <i
                class="fa fa-twitter"
              />
            </a>
            <a
              class="btn btn-link btn-floating btn-lg text-dark"
              data-mdb-ripple-color="dark"
              href="#!"
              role="button"
            >
              <i
                class="fa fa-facebook-f"
              />
            </a>
            <a
              class="btn btn-link btn-floating btn-lg text-dark"
              data-mdb-ripple-color="dark"
              href="#!"
              role="button"
            >
              <i
                class="fa fa-google-plus"
              />
            </a>
            <a
              class="btn btn-link btn-floating btn-lg text-dark"
              data-mdb-ripple-color="dark"
              href="#!"
              role="button"
            >
              <i
                class="fa fa-vimeo"
              />
            </a>
            <a
              class="btn btn-link btn-floating btn-lg text-dark"
              data-mdb-ripple-color="dark"
              href="#!"
              role="button"
            >
              <i
                class="fa fa-pinterest"
              />
            </a>
            <a
              class="btn btn-link btn-floating btn-lg text-dark"
              data-mdb-ripple-color="dark"
              href="#!"
              role="button"
            >
              <i
                class="fa fa-github"
              />
            </a>
          </section>
        </div>
      </footer>
    </div>
    ,
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
